File: Technology Platforms (Java, .NET)/Java (multithreading, recursive class, socket server, database, unit testing)/src/main/java/org/example/Child.java
package org.example;

import javax.persistence.*;

@Entity
public class Child {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String imie;
    private int wiek;

    // Relacja wiele do jednego - każde dziecko ma jednego rodzica.
    @ManyToOne
    @JoinColumn(name = "rodzic_id")
    private Person rodzic;

    public Child() {}

    public Child(String imie, int wiek) {
        this.imie = imie;
        this.wiek = wiek;
    }

    public Long getId() {
        return id;
    }

    public String getImie() {
        return imie;
    }

    public void setImie(String imie) {
        this.imie = imie;
    }

    public Person getRodzic() {
        return rodzic;
    }

    public int getWiek() {
        return wiek;
    }

    public void setRodzic(Person rodzic) {
        this.rodzic = rodzic;
    }

    @Override
    public String toString() {
        return "Child{" +
                "id=" + id +
                ", imie='" + imie + '\'' +
                '}';
    }
}

File: Technology Platforms (Java, .NET)/.NET (recursive class, LINQ, XML, and XPath, datagrid, multithreading)/Child.cs
using System.Xml.Serialization;

namespace WpfApp
{
    [XmlType("Child")]
    public class Child
    {
        public string Name { get; set; }

        public int Age { get; set; }

        public Child(string imie, int wiek)
        {
            Name = imie;
            Age = wiek;
        }

        public Child() {}

        public override string ToString()
        {
            return $"  ↳ {Name}, {Age} lat";
        }
    }
}
